src/model/Policy.lua

```diff
@@ -239,19 +239,25 @@ end
     * @return succeeds or not.
 ]]
 function Policy:addPolicies(sec, ptype, rules)
-    local size = #self.model[sec][ptype].policy
+    return self:addPoliciesWithAffected(sec, ptype, rules)~=0
+end
+
+--[[
+    * addPoliciesWithAffected adds policy rules to the model.
+    * @param sec the section, "p" or "g".
+    * @param ptype the policy type, "p", "p2", .. or "g", "g2", ..
+    * @param rules the policy rules.
+    * @return effected.
+]]
+function Policy:addPoliciesWithAffected(sec, ptype, rules)
+    local effected={}
     for _, rule in pairs(rules) do
         if not self:hasPolicy(sec, ptype, rule) then
-            table.insert(self.model[sec][ptype].policy, rule)
-            self.model[sec][ptype].policyMap[table.concat(rule,",")]=#self.model[sec][ptype].policy
+            table.insert(effected, rule)
+            self:addPolicy(sec, ptype, rule)
```

Comment:
Since this is checking hasPolicy twice (addPolicy also uses hasPolicy), can we do this instead? addPolicy returns true if the policy has been added.
```lua
for _, rule in pairs(rules) do
    if self:addPolicy(sec, ptype, rule) then
        table.insert(effected, rule)
    end
end
```

Reply:
Changed.